Feature/50 update frontend so that a displayname can be updated to an account

chenaicirt opened the pull request:
https://trello.com/c/PL6cXJMA/50-update-frontend-so-that-a-displayname-can-be-updated-to-an-account

chenaicirt committed:
able to see modal on the snackbar when the button in snackbar is clicked

chenaicirt committed:
edited the form so that we gather the relevant data

chenaicirt committed:
removed api services imports and functions that were from housing db and added in patchAccount method

chenaicirt committed:
added account model

chenaicirt committed:
used the api service route and linked it to the button click

chenaicirt committed:
modal seems to work now! Just need to wait for PR for account to be fixed

chenaicirt committed:
edited the html visually, and also edited the routes from some of the changes

chenaicirt committed:
updated the routes and interactions so that it would say displayname instead of username

chenaicirt committed:
minor changes so that snackbar and modal closes after submitting displayname

chenaicirt committed:
removed duplicate join page -- unsure where it came from

chenaicirt committed:
Merge branch 'develop' into feature/50-update-frontend-so-that-a-displayname-can-be-updated-to-an-account

zaclittleberry commented on src/app/_components/action-snack-bar-with-modal/snack-bar-component-example-snack.html:
```diff
@@ -0,0 +1,9 @@
+<button mat-raised-button
+  (click)="openDialog()"> Add Display Name </button>
+
+<!-- Copyright 2018 Google Inc. All Rights Reserved.
+    Use of this source code is governed by an MIT-style license that
+    can be found in the LICENSE file at http://angular.io/license -->
```
@chenaicirt is there a reason there is license text in this file in particular?

chenaicirt replied:
Nope, this was just taken from the snackbar demo code that was on the material website

chenaicirt replied:
should I delete this?

zaclittleberry commented on src/app/_services/api.service.ts:
```diff
@@ -49,25 +49,24 @@ export class ApiService {
-  getAccountLandlords() {
-    return this.httpClient.get<any>(`${this.apiUrl}/accounts/landlords`);
+  patchAccount(id, account: Account) {
+    return this.httpClient.patch<any>(`${this.apiUrl}/accounts/${id}`, account, {observe: 'response'});
```
the third param (object with observe response) is only needed if you need the full response (with headers) passed through. It doesn't look like you do, so this should be removed.

zaclittleberry reviewed (changes requested):
There's a number of things in this pr to change. I'm not going to do each individually for this one, so apply the following where appropriate:
The longer bit of text I gave you was meant to go in the modal, not in the snackbar. I don't remember what the snackbar said before but I think that was fine as it was.
Please rename things (functions, classes, services, etc) semantically. so AccountDialog, AccountSnackBar, etc. I can tell you lifted code from examples, which is fine, but the code needs to be changed to be appropriate for this codebase. Once it's changed you can also remove the license at the bottom of that one file as it shouldn't be necessary in this case.
Please stick to style conventions and indent code properly, and multiple words in file names should be separated with dashes. Also services have a convention of ending in .service.ts, and their class name ending in Service. It is best to stick to any conventions that already exist in a codebase.
Also, I'm pretty sure you imported some classes into files they weren't used in, and I think may have imported a class into the same file twice. Please clean up any code that isn't actively in use; the only things committed should be what's necessary for the ticket at hand to work.

chenaicirt committed:
updated/changed the folder names and file names

chenaicirt committed:
"some merge"
Merge branch 'feature/50-update-frontend-so-that-a-displayname-can-be-updated-to-an-account' of https://github.com/GazeDev/groups_app into feature/50-update-frontend-so-that-a-displayname-can-be-updated-to-an-account

chenaicirt committed:
removed the extra import of alert.service since we replaced that with the newly added alertaccount

chenaicirt committed:
updated to have - for multiword naming convention purposes

chenaicirt committed:
removed patch account extra argument

chenaicirt committed:
added a paragraph tag inside the dialog
